### PR TITLE
Set "min" and/or "step" value to input fields

### PR DESCRIPTION
**Pull request by fernandog**
@miigotu

### CODE REVIEW

**Comment by OmgImAlexis** [comment]
Before changing anything like this it's a good idea to check [caniuse.com](http://caniuse.com/#search=number), there's known issues with the number type.

**Comment by fernandog** [comment]
We already have that others fields. Just added to the missing ones.
What issue can cause?

**Comment by OmgImAlexis** [comment]
Check the known issues tab on the page I linked too.

**Comment by fernandog** [comment]
I did and didnt understand.

**Comment by OmgImAlexis** [comment]
Well this is a major one. "The .stepUp and .stepDown methods incorrectly invoke InvalidStateError exceptions in IE 10 and 11." This means we could end up with the inputs that use these not being able to be used with any js including ajax on IE10 and IE11.

**Comment by miigotu** [comment]
Need to change all of these to jquery ui spnner I think
https://jqueryui.com/spinner/

**Comment by fernandog** [comment]
But we dont use those in JS isnt?
Webserve reads directly from the component isnt?

File: gui/slick/views/config_providers.mako [hunk 1]
```diff
@@ -432,7 +432,7 @@ $('#config-components').tabs();
                             <label for="${curTorrentProvider.get_id()}_minleech">
                                 <span class="component-title" id="${curTorrentProvider.get_id()}_minleech_desc">Minimum leechers:</span>
                                 <span class="component-desc">
-                                    <input type="number" name="${curTorrentProvider.get_id()}_minleech" id="${curTorrentProvider.get_id()}_minleech" value="${curTorrentProvider.minleech}" class="form-control input-sm input75" />
+                                    <input type="number" min="1" step="1" name="${curTorrentProvider.get_id()}_minleech" id="${curTorrentProvider.get_id()}_minleech" value="${curTorrentProvider.minleech}" class="form-control input-sm input75" />
```
Comment (duramato) [inline gui/slick/views/config_providers.mako]: This should not be enforced

Comment (fernandog) [inline gui/slick/views/config_providers.mako]: We already do that for new installs. Default value

Comment (duramato) [inline gui/slick/views/config_providers.mako]: Default value is not min value

**Comment by miigotu** [comment]
huh? you apply that class on the input...

**Comment by fernandog** [comment]
Avoid negative numbers. Float numbers. Etc

File: gui/slick/views/config_providers.mako (hunk 1 above)
Comment (miigotu) [inline gui/slick/views/config_providers.mako]: default value for minleech is 0.

**Comment by OmgImAlexis** [comment]
@miigotu can you have a look through the caniuse page, @fernandog doesn't seem to understand it.

Basically the fields shouldn't be used until IE10 and IE11 support is added otherwise we could end up breaking functionality on there.

**Comment by miigotu** [comment]
@OmgImAlexis I will then, but I think ie10/11 just treat them as a text input (no fancy controls)

**Comment by Arcanemagus** [comment]
That's correct, when a browser doesn't understand the type given to an input element it defaults it to text, so you don't get the fancy controls and built in validation, but the end result sent back to the server is the exact same.

File: gui/slick/views/config_search.mako [hunk 2]
```diff
@@ -346,7 +346,7 @@
                                 <label>
                                     <span class="component-title">NZBget host:port</span>
                                     <span class="component-desc">
-                                        <input type="text" name="nzbget_host" id="nzbget_host" value="${sickbeard.NZBGET_HOST}" class="form-control input-sm input350" autocapitalize="off" />
+                                        <input type="text" min="1" step="1" name="nzbget_host" id="nzbget_host" value="${sickbeard.NZBGET_HOST}" class="form-control input-sm input350" autocapitalize="off" />
```
Comment (miigotu) [inline gui/slick/views/config_search.mako:349]: type="number"

Fix that and we can merge this for now and worry about jquery ui spinner later. This actually will make it easier to regex for replace.

**Comment by miigotu** [comment]
Will remake PR for you, cant use numeric for host =P

File: gui/slick/views/config_search.mako (hunk 2 above)
Comment (fernandog) [inline gui/slick/views/config_search.mako:349]: Sorry!  I thought it was only the port.

**Pull request closed without merging**